pkg/healthcheck/healthcheck.go

```diff
@@ -802,7 +802,14 @@ func getPodStatuses(pods []v1.Pod) map[string][]v1.ContainerStatus {
 	for _, pod := range pods {
 		if pod.Status.Phase == v1.PodRunning {
 			parts := strings.Split(pod.Name, "-")
-			name := strings.Join(parts[1:len(parts)-2], "-")
+			var name string
+			// All control plane pods  should have a name that results in 3 substrings string.Split on '-'
```

Comment:
controller pods should split into 4 parts because we recently added the `linkerd-` prefix (`linkerd-controller-b8c4c48c8-q85d2`)

in addition to checking on the length, i think it's safer to just ignore pods we don't recognize, consider something like:
```go
for _, pod := range pods {
	if pod.Status.Phase == v1.PodRunning && strings.HasPrefix(pod.Name, "linkerd-") {
		parts := strings.Split(pod.Name, "-")
		// All control plane pods should have a name that results in 4 substrings string.Split on '-'
		if len(parts) == 4 {
			name := parts[1]
			if _, found := statuses[name]; !found {
				statuses[name] = make([]v1.ContainerStatus, 0)
			}
			statuses[name] = append(statuses[name], pod.Status.ContainerStatuses...)
		}
	}
}
```

Reply:
@siggy Jinx! And not to bikeshed this to death, but some of our control plane pods will actually have 5 parts in their name, e.g. "linkerd-proxy-injector-5ffc675b8b-8zfrf".

There's a related issue as well, which is that if you install something else into your linkerd namespace and it's not healthy, that will cause this check to fail. So given that we're now using the "linkerd-" prefix everywhere, maybe we can fix both issues with something like:

```
	for _, pod := range pods {
		if pod.Status.Phase == v1.PodRunning {
			if strings.HasPrefix(pod.Name, "linkerd-") {
				parts := strings.Split(pod.Name, "-")
				// All control plane pods should have a name that results in at least 4
				// substrings when string.Split on '-'
				if len(parts) >= 4 {
					name := strings.Join(parts[1:len(parts)-2], "-")
					if _, found := statuses[name]; !found {
						statuses[name] = make([]v1.ContainerStatus, 0)
					}
					statuses[name] = append(statuses[name], pod.Status.ContainerStatuses...)
				}
			}
		}
	}
```

Reply:
@klingerf that all makes sense! i will continue this bikeshed to say we should stick with the single line `if` i had proposed, to decrease indent:
```
if pod.Status.Phase == v1.PodRunning && strings.HasPrefix(pod.Name, "linkerd-") {
```

Reply:
I like the Prefix check. It felt kinda strange checking for a pod that was not part of the components of the control plane.

Reply:
omg @siggy I didn't even see that your solution _already included_ the prefix check. nevermind me -- carry on.